File: ChainOfResponsibility/Go/chain/MsgProcessor.go
package chain

type MsgProcessor struct {
  msg string
  filter_chain *FilterChain
}

func NewMsgProcessor(msg string, filter_chain *FilterChain) *MsgProcessor {
  return &MsgProcessor{msg, filter_chain}
}

func (this *MsgProcessor) Process() string {
  return this.filter_chain.HandleFilter(this.msg)
}

File: ChainOfResponsibility/PHP/MsgProcessor.php
<?php
  class MsgProcessor {
    private $msg;
    private $filter_chain;

    public function __construct($msg, $filter_chain) {
      $this->msg = $msg;
      $this->filter_chain = $filter_chain;
    }

    public function process() {
      return $this->filter_chain->handleFilter($this->msg);
    }
  }
